### ISSUE TITLE
[Feature request] USB passthrough hotplug

### ISSUE THREAD

**Issue by sanjay900** [RPCS3/rpcs3#15074]
### Quick summary

It would be nice if USB device passthrough supported hotplug with passthrough. 

### Details

It would be nice if RPCS3 would detect a device that was plugged in mid-game and pass it through to the emulator, instead of needing to restart the game to pick up new devices.

It could either be implemented by constantly listing devices in the usb device thread  or we could actually hook into libusbs detection routines for macos + linux and hook into windows' device detection for device detection there.

**Comment by DoubleSpicy** [comment]
Constantly listing usb device list can be performance-hitting, 

Instead of hooking directly we might be able to leverage libusb's `device hotplug event notification` callback function.
https://libusb.sourceforge.io/api-1.0/libusb_hotplug.html

**Comment by sanjay900** [comment]
> Constantly listing usb device list can be performance-hitting,
> 
> Instead of hooking directly we might be able to leverage libusb's `device hotplug event notification` callback function. https://libusb.sourceforge.io/api-1.0/libusb_hotplug.html

Yeah thats what we would like to do, but that callback is not yet implemented for windows.

**Comment by doctorduck2000** [comment]
This would be really convenient for stuff like the Skylanders portal, I hope this gets added.

**Comment by sanjay900** [comment]
Sadly that hotplug stuff still hasn't landed in libusb and it doesn't really look like it will land any time soon.

**Comment by DoubleSpicy** [comment]
An alternative for now here maybe is to poll usb devices at low frequency for windows, and implement seperately w/ libusb Hotplug for macos/linux.

**Comment by sanjay900** [comment]
That makes sense

**Comment by sanjay900** [comment]
With hotplug, would we disconnect a virtual device if the real device is plugged in?
Say if you plug in a turntable, do we then disable an emulated turntable, or just leave the emulated turntables connected.

**Comment by AniLeo** [comment]
Turntables are not handled by the game through cellUsbd requiring passthrough of the USB device to the game, they're pads so they're handled by cellPad

**Comment by sanjay900** [comment]
Guitar hero guitars are handled as pads by the games
Turntables and Rock band guitars are handled via cellUsbd.
I can pass through a physical turntable and a physical rock band guitar just fine
But that doesn't work for guitar hero guitars.
